packages/sources/lotus/src/transport/balance.ts

```diff
@@ -0,0 +1,141 @@
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/balance'
+import { BigNumber } from 'ethers'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+
+export type BalanceTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: {
+      method: string
+      params: string[]
+      id: number
+      jsonrpc: string
+    }
+  }
+}
+
+type RequestParams = typeof inputParameters.validated
+
+export class TotalBalanceTransport extends SubscriptionTransport<BalanceTransportTypes> {
+  requester!: Requester
+  settings!: BalanceTransportTypes['Settings']
+  endpointName!: string
+
+  async initialize(
+    dependencies: TransportDependencies<BalanceTransportTypes>,
+    _adapterSettings: BalanceTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, _adapterSettings, _endpointName, transportName)
+    this.requester = dependencies.requester
+    this.settings = _adapterSettings
+    this.endpointName = _endpointName
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<BalanceTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    if (!entries.length) {
+      await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+      return
+    }
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
```

Comment:
Like the other adapters, I think we'll want to sleep after all bg cycles
```suggestion
    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
```